tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -150,14 +201,21 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
 		ginkgo.By("Verify that the created CNS volumes are compliant and have correct policy id")
+		var volumeID string
 		for i, pv := range pvs {
-			volumeID := pv.Spec.CSI.VolumeHandle
-			storagePolicyMatches, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyNames[i])
+			volumeID = pv.Spec.CSI.VolumeHandle
+			if guestCluster {
+				volumeID = getVolumeIDFromSupervisorCluster(pv.Spec.CSI.VolumeHandle)
+				gomega.Expect(volumeID).NotTo(gomega.BeEmpty())
+			}
+			storagePolicyExists, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyNames[i])
 			e2eVSphere.verifyVolumeCompliance(volumeID, true)
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
-			gomega.Expect(storagePolicyMatches).To(gomega.BeTrue(), "storage policy verification failed")
-			gomega.Expect(e2eVSphere.verifyDatastoreMatch(volumeID, sharedvmfsURL)).To(
-				gomega.BeTrue(), "volume %v was created on wrong ds", sharedvmfsURL)
+			gomega.Expect(storagePolicyExists).To(gomega.BeTrue(), "storage policy verification failed")
+			dsUrlWhereVolumeIsPresent := findDatastoreWhereVolumeIsPresent(e2eVSphere, volumeID)
+			framework.Logf("Verify if VolumeID: %s is created on the datastore: %s", volumeID, sharedvmfsURL)
+			verifyVolumePresentOnGivenDatastore(dsUrlWhereVolumeIsPresent, []string{sharedvmfsURL})
+
```

Comment:
you can use the existing code itself here right

Reply:
Done